src/data-model-providers/codegen/CodegenDataModelProvider.cpp

```diff
@@ -159,6 +159,16 @@ std::optional<DataModel::ActionReturnStatus> CodegenDataModelProvider::Invoke(co
                                                                               TLV::TLVReader & input_arguments,
                                                                               CommandHandler * handler)
 {
+    // Ensure the command actually exists on the relevant cluster instance.
```

Comment:
This does not in fact ensure that.  This is ensuring the _cluster_ exists.  It does not ensure that the cluster instance involved exposes this command.  That would involve interrogating the CHI instance to see what commands it supports for this specific cluster path.

Note that just doing InvokeCommand on the CHI might not do that check, because the invariant so far has been that InvokeCommand is only called if you claim to have the command at all....

Reply:
Updated comment, I think in this case I would say "this makes it better than before" and would follow up on changing things.

From a processing logic perspective, I think we should not `require` invoke to accept invalid paths, but not preventing them from double-checking them either. I am a fan of `trust but verify` in this case. 

The current contract in CHI is that commands that are not handled are silently skipped (to allow for ember fallback) so it `should` handle commands that are unknown. However actually double checking this is probably not practical.